sale_timesheet_rounded/models/account_analytic_line.py

```diff
@@ -34,39 +34,37 @@ def _init_column(self, column_name):
 
     @api.onchange('unit_amount')
     def _onchange_unit_amount(self):
-        self.with_context(force_compute=1)._update_unit_amount_rounded()
+        self.unit_amount_rounded = self.with_context(
+            force_compute=1)._calc_unit_amount_rounded()
 
-    def _update_unit_amount_rounded(self):
+    def _calc_unit_amount_rounded(self):
+        self.ensure_one()
         force_compute = self.env.context.get('force_compute')
         if self.env.context.get('grid_adjust'):
             # TODO: do we need this?
             # support timesheet_grid from EE version
             # force_compute if the timesheet is change in the grid view
             force_compute = True
-        for rec in self.with_context(
-            timesheet_rounding=True,
-            # avoid recursion when writing from this method
-            update_unit_amount_rounded=True,
-        ):
-            if not rec.project_id:
-                # no project: do nothing
-                continue
-            if rec.product_id and rec.product_id.expense_policy != 'no':
-                # expense line: do nothing
-                continue
-            if rec.unit_amount_rounded and not force_compute:
-                # already set, no forcing: do nothing
-                continue
-            if rec.project_id.timesheet_rounding_method == 'NO':
-                # No rounding to apply: copy the value
-                rec.unit_amount_rounded = rec.unit_amount
-            else:
-                rec.unit_amount_rounded = self._calc_rounded_amount(
-                    rec.project_id.timesheet_rounding_unit,
-                    rec.project_id.timesheet_rounding_method,
-                    rec.project_id.timesheet_rounding_factor,
-                    rec.unit_amount,
-                )
+        # TODO: do we still need this?
+        self = self.with_context(timesheet_rounding=True)
+        project_rounding = (
+            self.project_id and
+            self.project_id.timesheet_rounding_method != 'NO'
+        )
+        expense_line = (
+            self.product_id and self.product_id.expense_policy != 'no'
```

Comment:
Here we shouldn't test if the product has an expense policy but check if the analytic line is a timesheet one or anything else.